btk/survey.py

```diff
@@ -20,14 +22,95 @@
         "extinction",  # Exponential extinction coefficient for atmospheric absorption.
     ],
 )
 
+
+def get_psf(mirror_diameter, effective_area, filt_wavelength, fwhm, atmospheric_model="Kolmogorov"):
+    """Credit: WeakLensingDeblending (https://github.com/LSSTDESC/WeakLensingDeblending)
+    Defines a synthetic galsim PSF model
+    Args:
+        mirror_diameter (float): in meters [m]
+        effective_area (float): effective total light collecting area in square meters [m2]
+        filt_wavelength (string): filter wavelength
+        fwhm (float): fwhm of the atmospheric component
+        atmospheric_model (string): type of atmospheric model
+    Returns:
+        psf_model: galsim psf model
+    """
+
+    # define atmospheric psf
+    if atmospheric_model == "Kolmogorov":
+        atmospheric_psf_model = galsim.Kolmogorov(fwhm=fwhm)
+    elif atmospheric_model == "Moffat":
+        atmospheric_psf_model = galsim.Moffat(2, fwhm=fwhm)
+    elif atmospheric_model is None:
+        atmospheric_psf_model = None
+    else:
+        raise NotImplementedError(
+            f"The atmospheric model request '{atmospheric_model}' is incorrect or not implemented."
+        )
+
+    # define optical psf if available
+    if mirror_diameter > 0:
+        mirror_area = np.pi * (0.5 * mirror_diameter) ** 2
+        area_ratio = effective_area / mirror_area
+        if area_ratio <= 0 or area_ratio > 1:
+            raise RuntimeError(
+                "Incompatible effective-area and mirror-diameter values."
+            )
+        obscuration_fraction = np.sqrt(1 - area_ratio)
+        lambda_over_diameter = 3600 * np.degrees(
+            1e-10 * filt_wavelength / mirror_diameter
+        )
+        optical_psf_model = galsim.Airy(
+            lam_over_diam=lambda_over_diameter, obscuration=obscuration_fraction
+        )
+    else:
+        optical_psf_model = None
+
+    # define the psf model according to the components we have
+    if isinstance(atmospheric_psf_model, galsim.GSObject) and isinstance(optical_psf_model, galsim.GSObject):
+        psf_model = galsim.Convolve(atmospheric_psf_model, optical_psf_model)
+    elif isinstance(atmospheric_psf_model, galsim.GSObject) and optical_psf_model is None:
+        psf_model = atmospheric_psf_model
+    elif atmospheric_psf_model is None and isinstance(optical_psf_model, galsim.GSObject):
+        psf_model = optical_psf_model
+    elif atmospheric_psf_model is None and optical_psf_model is None:
+        raise RuntimeError(
+            f"Neither the atmospheric nor the optical PSF components are defined."
+        )
+    
+    return psf_model.withFlux(1.0)
+
+
+def get_custom_psf(psf_dir):
+    """Generates a custom PSF galsim model from FITS file(s)
+    Args:
+        psf_dir (string): directory where the PSF FITS files are
+    Returns:
+        psf_model: galsim PSF model
+    """
+
+    psf_files = os.listdir(psf_dir)
+    if len(psf_files)>1:
+        psf_file = rd.choice(psf_files)
```

Comment:
I think it would be nice to add an option to specify a seed to make the choice of PSF reproducible.

Reply:
Thanks @HironaoMiyatake, I agree. This is related to #69 and it would be great to make this consistent throughout BTK to ensure galaxy blend + psf reproducibility. I would probably leave this for a separate PR that closes #69 but this is definitely important.

Reply:
I agree. Let's close this PR and work on reproducibility in a PR for #69.